pkg/blocks/subttp.go

```diff
@@ -21,33 +21,51 @@ package blocks
 
 import (
 	"errors"
-	"fmt"
 	"strings"
 
 	"github.com/facebookincubator/ttpforge/pkg/logging"
-	"gopkg.in/yaml.v3"
 )
 
 // SubTTPStep represents a step within a parent TTP that references a separate TTP file.
 type SubTTPStep struct {
-	*Act    `yaml:",inline"`
 	TtpFile string            `yaml:"ttp"`
 	Args    map[string]string `yaml:"args"`
 
-	// Omitting because the sub steps will contain the cleanups.
-	CleanupSteps []CleanupAct `yaml:"-,omitempty"`
-	ttp          *TTP
-	subExecCtx   TTPExecutionContext
+	ttp                   *TTP
+	subExecCtx            TTPExecutionContext
+	firstStepToCleanupIdx int
 }
 
 // NewSubTTPStep creates a new SubTTPStep and returns a pointer to it.
 func NewSubTTPStep() *SubTTPStep {
 	return &SubTTPStep{}
 }
 
-// GetCleanup returns a slice of CleanupAct associated with the SubTTPStep.
-func (s *SubTTPStep) GetCleanup() []CleanupAct {
-	return []CleanupAct{s}
+type subTTPCleanupAction struct {
+	actionDefaults
+	step *SubTTPStep
+}
+
+func (a *subTTPCleanupAction) Execute(execCtx TTPExecutionContext) (*ActResult, error) {
+	cleanupResults, err := a.step.ttp.startCleanupAtStepIdx(a.step.firstStepToCleanupIdx, &execCtx)
```

Comment:
simplified this and made cleanup logic work with new paradigm - subttps are a special tricky case

Reply:
New E2E test covers nuanced scenarios (SubTTP fails, and needs to be partially cleaned up)